### PR TITLE
feat(a11y): added header, main, footer aria landmarks

### PR DESCRIPTION
**Pull request by jeabakker**
ref #5087

### CODE REVIEW

File: mod/members/languages/en.php [hunk 1]
```diff
@@ -9,6 +9,7 @@
 	'members:label:online' => 'Online',
 	'members:label:search' => 'Search results',
 	'members:search' => 'Search members',
+	'members:aria:label:user_search' => "Search for users",
```
Comment (jdalsem) [inline mod/members/languages/en.php:12]: should it not be search for members? than also update the label key

File: mod/members/views/default/members/sidebar.php [hunk 2]
```diff
@@ -3,12 +3,12 @@
...
+$body = elgg_view_form('members/search', [
 	'method' => 'get',
-	'action' => 'members/search',
+	'action' => elgg_generate_url('search:user:user'),
 	'disable_security' => true,
-];
-
-$body = elgg_view_form('members/search', $params);
+	'role' => 'search',
+	'aria-label' => elgg_echo('members:aria:label:user_search'),
```
Comment (jdalsem) [inline mod/members/views/default/members/sidebar.php:11]: label member_search

File: mod/pages/languages/en.php [hunk 3]
```diff
@@ -19,6 +19,7 @@
 	'collection:object:page:group' => "Group pages",
 	'add:object:page' => "Add a page",
 	'edit:object:page' => "Edit this page",
+	'menu:pages_nav:header' => "Sub pages",
```
Comment (jdalsem) [inline mod/pages/languages/en.php:22]: Sub-pages or subpages

File: mod/search/languages/en.php [hunk 4]
```diff
@@ -12,4 +12,6 @@
 	'search:comment_on' => 'Comment on "%s"',
 	'search:unknown_entity' => 'Unknown Entity type',
 	'search:empty_query' => 'Please provide a valid search query',
+	
+	'search:aria:label:site_search' => "Search on the site",
```
Comment (jdalsem) [inline mod/search/languages/en.php:16]: 'search:aria:label:site'? double search

File: views/default/admin/users/header.php [hunk 5]
```diff
@@ -33,6 +33,8 @@
 	'action' => elgg_get_current_url(),
 	'disable_security' => true,
 	'class' => $search_class,
+	'role' => 'search',
+	'aria-label' => elgg_echo('aria:label:admin:user:search'),
```
Comment (jdalsem) [inline views/default/admin/users/header.php:37]: should it not be `admin:users:search`? extra s

File: views/default/forms/admin/users/search.php [hunk 6]
```diff
@@ -15,9 +15,10 @@
 	'#type' => 'fieldset',
 	'fields' => [
 		[
-			'#type' => 'text',
+			'#type' => 'search',
 			'#class' => 'elgg-field-stretch',
 			'name' => 'q',
+			'aria-label' => elgg_echo('search'), // because we don't have a #label
 			'placeholder' => elgg_echo('search'),
```
Comment (jdalsem) [inline views/default/forms/admin/users/search.php:22]: should we magically default aria-label to the placeholder? this feels like a lot of duplication

File: docs/appendix/upgrade-notes/5.0-to-5.1.rst
Comment (jdalsem) [inline docs/appendix/upgrade-notes/5.0-to-5.1.rst:1]: Maybe write something about the changes in document element structure DIVS -> native elements (HEADER/FOOTER/MAIN) etc

**Review by jdalsem — CHANGES_REQUESTED** [review]

File: mod/blog/languages/en.php [hunk 7]
```diff
@@ -16,6 +16,7 @@
 	'edit:object:blog' => 'Edit blog post',
 	'notification:object:blog:publish' => "Send a notification when a blog is published",
 	'notifications:mute:object:blog' => "about the blog '%s'",
+	'menu:block_archive:header' => "Blog archive",
```
[the remark this answers is not in the document]
Comment (jeabakker) [reply mod/blog/languages/en.php:19]: fixed

File: mod/members/languages/en.php (hunk 1 above)
Comment (jeabakker) [reply mod/members/languages/en.php:12]: fixed

File: mod/members/views/default/members/sidebar.php (hunk 2 above)
Comment (jeabakker) [reply mod/members/views/default/members/sidebar.php:11]: fixed

File: mod/pages/languages/en.php (hunk 3 above)
Comment (jeabakker) [reply mod/pages/languages/en.php:22]: fixed

File: mod/search/languages/en.php (hunk 4 above)
Comment (jeabakker) [reply mod/search/languages/en.php:16]: not going to change

File: views/default/admin/users/header.php (hunk 5 above)
Comment (jeabakker) [reply views/default/admin/users/header.php:37]: fixed

File: docs/appendix/upgrade-notes/5.0-to-5.1.rst
Comment (jeabakker) [reply docs/appendix/upgrade-notes/5.0-to-5.1.rst:1]: fixed

**Review by jdalsem — APPROVED** [review]

**Pull request merged**